qiskit/primitives/backend_sampler.py

```diff
@@ -0,0 +1,252 @@
+        return self
+
+    @property
+    def transpile_options(self) -> Options:
+        """Return the transpiler options for transpiling the circuits."""
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields) -> BackendSampler:
+        """Set the transpiler options for transpiler.
+        Args:
+            **fields: The fields to update the options.
+        Returns:
+            self.
+        Raises:
+            QiskitError: if the instance has been closed.
+        """
+        self._check_is_closed()
+
+        self._transpile_options.update_options(**fields)
+        return self
+
+    def _call(
+        self,
+        circuits: Sequence[int],
+        parameter_values: Sequence[Sequence[float]],
+        **run_options,
+    ) -> SamplerResult:
+        self._check_is_closed()
+
+        # This line does the actual transpilation
+        transpiled_circuits = self.transpiled_circuits
+        bound_circuits = [
+            transpiled_circuits[i]
+            if len(value) == 0
+            else transpiled_circuits[i].bind_parameters((dict(zip(self._parameters[i], value))))
+            for i, value in zip(circuits, parameter_values)
+        ]
+        bound_circuits = self._bound_pass_manager_run(bound_circuits)
+
+        # Run
+        run_opts = copy.copy(self.run_options)
+        run_opts.update_options(**run_options)
+        result = self._backend.run(bound_circuits, **run_opts.__dict__).result()
+
+        return self._postprocessing(result, bound_circuits)
+
+    def close(self):
+        self._is_closed = True
+
+    def _postprocessing(self, result: "Result", circuits: list["QuantumCircuit"]) -> SamplerResult:
+
+        counts = result.get_counts()
+        if not isinstance(counts, list):
+            counts = [counts]
+
+        shots = sum(counts[0].values())
+
+        probabilies = []
+        metadata: list[dict[str, Any]] = [{}] * len(circuits)
+        for count in counts:
+            prob_dist = {k: v / shots for k, v in count.int_outcomes().items()}
+            probabilies.append(QuasiDistribution(prob_dist))
+            for metadatum in metadata:
+                metadatum["shots"] = shots
+        return SamplerResult(probabilies, metadata)
+
+    def _transpile(self):
+        from qiskit.compiler import transpile
+
+        self._transpiled_circuits = cast(
+            "list[QuantumCircuit]",
+            transpile(
+                self.preprocessed_circuits,
+                self.backend,
+                **self.transpile_options.__dict__,
+            ),
+        )
+
+    def _check_is_closed(self):
+        if self._is_closed:
+            raise QiskitError("The primitive has been closed.")
+
+    def _bound_pass_manager_run(self, circuits):
+        if self._bound_pass_manager is None:
+            return circuits
+        else:
+            return cast("list[QuantumCircuit]", self._bound_pass_manager.run(circuits))
+
+    @staticmethod
+    def result_to_dict(
```

Comment:
This method is not necessary for terra. It's only for the internal version.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8668/commits/daa2b177bcd4a6648b265ed67a3d6d5fddeb5c2f